src/Optimizer/Transformer/DetermineHeroImages.php

```diff
@@ -59,7 +60,14 @@ final class DetermineHeroImages implements Transformer {
 	 *
 	 * @var string
 	 */
-	const INITIAL_CONTENT_IMAGE_XPATH_QUERY = './*[1]//amp-img[ not( @data-hero ) ]';
+	const INITIAL_CONTENT_IMAGE_XPATH_QUERY = "./*[1]//amp-img[ not( @data-hero ) ] | ./div[ contains( @class, 'has-parallax' ) and ( contains( @data-amp-original-style, 'background-image' ) or contains( @style, 'background-image' ) ) ]";
+
+	/**
+	 * Regular expression pattern to extract the URL from a CSS background-image property.
+	 *
+	 * @var string
+	 */
+	const CSS_BACKGROUND_IMAGE_URL_REGEX_PATTERN = '/background-image\s*:\s*url\(\s*(?<url>[^)]*\s*)/i';
```

Comment:
Should we care about potential quotes around the URL? E.g.:

- `url("http://www.example.com/sample.jpg")`
- `url('http://www.example.com/sample.jpg')`
- `url(http://www.example.com/sample.jpg)`

Here's a slightly modified pattern that could cover the cases with quotes

```regex
/background-image\s*:\s*url\(\s*['"]?(?<url>.*?)['"]?\s*\)/i
```